api/src/chat/validation-rules/is-channel-data.ts

```diff
@@ -12,18 +12,21 @@ import {
   ValidatorConstraint,
   ValidatorConstraintInterface,
 } from 'class-validator';
+import { z } from 'zod';
 
-export function isChannelData(channel: any) {
-  return (
-    typeof channel === 'object' &&
-    channel.name &&
-    typeof channel.name === 'string'
-  );
+export const channelSchema = z.object({
+  name: z.string(),
```

Comment:
I think we should do a regex check to ensure that the name ends with "-channel". WDYT?

Reply:
Can we move the schema in the type file, do we have a "types/channel.ts" ?